File: app/src/main/java/com/yanxing/ui/NetworkLibraryFragment.java
package com.yanxing.ui;

import android.widget.TextView;

import com.yanxing.base.BaseFragment;
import com.yanxing.networklibrary.RetrofitManage;
import com.yanxing.networklibrary.Transformer;
import com.yanxing.networklibrary.AbstractObserver;
import com.yanxing.ui.retrofit.DouBanAPI;
import com.yanxing.model.DouBan;

import butterknife.BindView;

/**
 * Networklibrary 例子
 * Created by lishuangxiang on 2016/5/9.
 */
public class NetworkLibraryFragment extends BaseFragment {

    @BindView(R.id.data)
    TextView mData;

    private String mBaseUrl = "https://api.douban.com/v2/movie/";

    @Override
    protected int getLayoutResID() {
        return R.layout.fragment_rxjava;
    }

    @Override
    protected void afterInstanceView() {
        RetrofitManage.getInstance().init(mBaseUrl,true);
        getData();
    }

    /**
     * 获取电影排行
     */
    public void getData() {
        RetrofitManage.getInstance().getRetrofit().create(DouBanAPI.class)
                .getTopMovie(0,10)
                .compose(new Transformer<DouBan>().iOMainHasProgress(this,getFragmentManager(),"请稍等..."))
                .subscribe(new AbstractObserver<DouBan>(getActivity(),getFragmentManager(),false) {
                    @Override
                    public void onCall(DouBan douBan) {

                    }

                    @Override
                    public void onNext(DouBan douBan) {
                        super.onNext(douBan);
                        StringBuilder stringBuilder = new StringBuilder();
                        for (int i = 0; i < douBan.getSubjects().size(); i++) {
                            stringBuilder.append(String.valueOf(i));
                            stringBuilder.append(".");
                            stringBuilder.append(douBan.getSubjects().get(i).getTitle());
                            stringBuilder.append("\n");
                        }
                        mData.setText(stringBuilder.toString());
                    }
                });
    }
}

File: app/src/main/java/com/yanxing/ui/NetworkLibraryFragment.kt
package com.yanxing.ui

import com.tencent.bugly.Bugly.applicationContext
import com.yanxing.baselibrary.BaseFragment
import com.yanxing.model.DouBan
import com.yanxing.networklibrary.BaseAbstractObserver
import com.yanxing.networklibrary.RetrofitManage
import com.yanxing.networklibrary.Transformer
import com.yanxing.ui.retrofit.DouBanAPI
import kotlinx.android.synthetic.main.fragment_networklibrary_kt.*

/**
 * Networklibrary kotlin版例子
 * @author 李双祥 on 2018/3/2.
 */
class NetworkLibraryFragment : BaseFragment() {

    val mBaseUrl = "https://api.douban.com/v2/movie/"

    override fun getLayoutResID(): Int {
        return R.layout.fragment_networklibrary_kt
    }

    override fun afterInstanceView() {
        RetrofitManage.getInstance().init(mBaseUrl, true)
        getData()
    }

    /**
     * 获取电影排行
     */
    fun getData() {
        RetrofitManage.getInstance().retrofit.create(DouBanAPI::class.java)
                .getTopMovie(0, 10)
                .compose(Transformer<DouBan>().iOMainHasProgress(this, mFragmentManager, "请稍等..."))
                .subscribe(object : BaseAbstractObserver<DouBan>(applicationContext, mFragmentManager) {
                    override fun onCall(douBan: DouBan) {
                        val stringBuilder = StringBuilder()
                        for (i in 0 until douBan.subjects.size) {
                            stringBuilder.append(i.toString())
                            stringBuilder.append(".")
                            stringBuilder.append(douBan.subjects[i].title)
                            stringBuilder.append("\n")
                        }
                        content.setText(stringBuilder.toString())
                    }
                })
    }
}